modules/@apostrophecms/page/index.js

```diff
@@ -1276,11 +1276,14 @@ database.`);
             if (parent._id !== oldParent._id) {
               const matchOldParentSlugPrefix = new RegExp('^' + self.apos.util.regExpQuote(self.apos.util.addSlashIfNeeded(oldParent.slug)));
               if (moved.slug.match(matchOldParentSlugPrefix)) {
-                let slugStem = parent.slug;
-                if (slugStem !== '/') {
-                  slugStem += '/';
-                }
-                moved.slug = moved.slug.replace(matchOldParentSlugPrefix, self.apos.util.addSlashIfNeeded(parent.slug));
+                const movedSlugCandidate = moved.slug
+                  .split('/')
+                  .slice(0, -1)
+                  .join('/');
+
+                moved.slug = parent.slug.endsWith(movedSlugCandidate)
```

Comment:
Also I think this could be confused in the following scenario:

These are peers:

```
/foo
/foobar
```

(Note no second slash)

And the second one gets moved under the first.

That should result in:

/foo/foobar

(Because no slash)

Tests should cover this case.

Reply:
This is in addition to the concern about testing plain old:

/foo
/bar

To make sure that when /bar is moved under /foo we still get /foo/bar in that typical case.